front: handle uncaught exceptions for e2e tests

Maymanaf opened the pull request:
Closes #8883

Only test 008 is excluded from this PR due to multiple exceptions causing it to fail. We're waiting for fixes from issues #11257 and #11256

emersion commented on front/tests/pages/stdcm/via-section.ts:
```diff
@@ -117,6 +120,7 @@ class ViaSection extends STDCMPage {
         await this.getViaType(viaNumber).selectOption(SERVICE_STOP);
         await expect(this.getViaStopTime(viaNumber)).toHaveValue(serviceStop.default);
         await this.getViaStopTime(viaNumber).fill(serviceStop.input);
+        await this.page.waitForTimeout(EXPLICIT_UI_STABILITY_TIMEOUT); // Wait for the animation to complete
```
Could we replace this with `scrollIntoViewIfNeeded()`, which waits for the element to be stable? Or listen for the `animationend` event?

Maymanaf committed:
front: handle uncaught exceptions for e2e tests

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

front: restructure and fix timetable e2e test

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

front: fix stdcm e2e test

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

front: fix project e2e test

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

front: handle undefined values in transformBoundariesDataToRangesData

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

SharglutDev commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -159,7 +159,7 @@ const SimulationResults = ({
 
   return (
-    <div className="simulation-results">
+    <div data-testid="simulation-results" className="simulation-results">
```
Is this data-testid used ?

Maymanaf replied:
it's not but nothing wrong with having the data-testid 
should I remove it ?

SharglutDev commented on front/tests/008-timetable-item.spec.ts:
For the second commit "fix timetable test", would you mind to add a description in the commit with the reason why they needed a fix, it's not really clear for me

Maymanaf replied:
The commit handles some uncaught exceptions related to the timetable.
I will squash all the commits into the first one

SharglutDev commented on front/tests/014-stdcm-linked-train.spec.ts:
```diff
@@ -101,19 +101,19 @@ test.describe('Verify stdcm simulation page', () => {
 
   /** *************** Test 2 **************** */
   test('Verify STDCM posterior linked train', async ({ page: _ }, testInfo) => {
+    await originSection.fillOriginDetailsLight(undefined, 'respectDestinationSchedule', true);
+    await linkedTrainSection.posteriorLinkedPathDetails();
+    await viaSection.fillAndVerifyViaDetails({
+      viaNumber: 1,
+      ciSearchText: 'mid_east',
+    });
```
Same as the previous comment. A description in the 3rd commit of what was the bug and why the changes fixes it. Does moving this block of tests fixes something ? Maybe we should also leave a comment here in the code if it is so someone doesn't move it again later. 
I find it a bit not logic to test first the itinerary part and not the consist one. In most of the case, a user will first fill the consist form and then the itinerary one.

Maymanaf replied:
The commit handles some uncaught exceptions related to stdcm
I will squash all the commits into the first one

SharglutDev commented on front/tests/002-project-management.spec.ts:
Also here an explanation fr the 4th commit of why it was broken with the previous method and not this one plz :)

SharglutDev commented on front/src/applications/operationalStudies/utils.ts:
```diff
@@ -79,26 +79,33 @@ export const transformBoundariesDataToPositionDataArray = <T extends 'gradient'
   // TODO : remove electrical profiles
-  const formatedData = boundariesData.boundaries.map((boundary, index) => ({
-    start: index === 0 ? 0 : mmToM(boundariesData.boundaries[index - 1]),
+  if (
+    !boundariesData ||
+    !Array.isArray(boundariesData.boundaries) ||
+    !Array.isArray(boundariesData.values)
+  ) {
+    return [];
+  }
+
+  const { boundaries, values } = boundariesData;
+
+  const formattedData: ElectricalRangesData<T>[] = boundaries.map((boundary, index) => ({
+    start: index === 0 ? 0 : mmToM(boundaries[index - 1]),
```
Do these changes fixes the [bug](https://github.com/OpenRailAssociation/osrd/issues/11256) or just hide it ? If you found a explanation of the issue, could you update its description so its easier to fix ?

Maymanaf replied:
It doesn't fix the bug, but it does fix an uncaught exception caused by undefined values in `transformBoundariesDataToRangesDat`a

SharglutDev reviewed (approved):
Lgtm and tested, thanks for the PR !

emersion commented on front/src/applications/operationalStudies/utils.ts:
```diff
@@ -79,26 +79,33 @@ export const transformBoundariesDataToPositionDataArray = <T extends 'gradient'
+  if (
+    !boundariesData ||
+    !Array.isArray(boundariesData.boundaries) ||
+    !Array.isArray(boundariesData.values)
```
These two conditions can't be possible according to the type system.

emersion commented on front/src/applications/operationalStudies/utils.ts:
```diff
@@ -79,26 +79,33 @@ export const transformBoundariesDataToPositionDataArray = <T extends 'gradient'
-  const formatedData = boundariesData.boundaries.map((boundary, index) => ({
-    start: index === 0 ? 0 : mmToM(boundariesData.boundaries[index - 1]),
+  if (
+    !boundariesData ||
```
Would be cleaner to handle this in the caller: `transformBoundariesDataToRangesData(electrifications ?? [])`

emersion commented:
Sorry, had some more comments!

emersion reviewed (approved):
Thanks!
